tokio/src/fs/file.rs

```diff
@@ -394,6 +394,53 @@ impl File {
         Ok(File::from_std(std_file))
     }
 
+    /// Destructures `File` into a [`std::fs::File`][std].
+    ///
+    /// # Examples
+    ///
+    /// ```no_run
+    /// use tokio::fs::File;
+    ///
+    /// # async fn dox() -> std::io::Result<()> {
+    /// let tokio_file = File::open("foo.txt").await?;
+    /// let std_file = file.into_std().await;
+    /// # Ok(())
+    /// # }
+    /// ```
+    pub async fn into_std(mut self) -> sys::File {
+        self.complete_inflight().await;
+        let std_file = Arc::downcast::<sys::File>(self.std).expect("Arc::downcast failed");
```

Comment:
Why is the downcast needed here?

Reply:
TBH I took your suggestion from the issue. I guess I can directly `Arc::try_unwrap(std_file).expect("Arc::try_unwrap failed")` after awaiting to complete inflight operations